external/mcpclient/src/mcp-client-plugin.ts

```diff
@@ -189,8 +160,16 @@ export class McpClientPlugin implements ChatPromptPlugin<'mcpClient', McpClientP
     }
   }
 
-  private makeMcpClientPlugin(serverUrl: string) {
-    const transport = this.createTransport(serverUrl);
+  private async makeMcpClientPlugin(
+    serverUrl: string,
+    headers: ValueOrFactory<Record<string, string>> | undefined
```

Comment:
nit: prefer `headers?: ValueOrFactory<Record<string, string>>`

Reply:
Hm, I think i prefer this one, because then it's a clear decision that `headers` are not required (I actually forgot in a few places so I changed it to this). Lmk if you disagree!

Reply:
if thats the case then I would consider using `null` like you did on line 41 for consistency. Non blocking tho.